http/handler.go

```diff
@@ -303,6 +303,19 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 		r = r.WithContext(ctx)
 		r = r.WithContext(namespace.ContextWithNamespace(r.Context(), namespace.RootNamespace))
 
+		// Set some response headers with raft node id (if applicable) and hostname, if available
+		nodeID := core.GetRaftNodeID()
+		if nodeID != "" {
+			w.Header().Set("X-Vault-Raft-Node-ID", nodeID)
+		}
+
+		// Swallow this error since we don't want to pollute the logs and we also don't want to
+		// return an HTTP error here. This information is best effort.
+		hostname, _ := os.Hostname()
```

Comment:
This adds a system call to every API request. Can we do this once outside the handler and use a cached value? Vault will need to restart to pickup a hostname change, but i think that's appropriate.

Reply:
Yep, sounds good